svc-licenses/licenses/licenses.go

```diff
@@ -103,3 +112,144 @@ func (e *ExternalInterface) GetLicenseResource(req *licenseproto.GetLicenseResou
 	resp.StatusCode = http.StatusOK
 	return resp
 }
+
+// InstallLicenseService to install license
+func (e *ExternalInterface) InstallLicenseService(req *licenseproto.InstallLicenseRequest) response.RPC {
+	var resp response.RPC
+	var contactRequest model.PluginContactRequest
+	var installreq dmtf.LicenseInstallRequest
+
+	genErr := JsonUnMarshalFunc(req.RequestBody, &installreq)
+	if genErr != nil {
+		errMsg := "Unable to unmarshal the install license request" + genErr.Error()
+		log.Error(errMsg)
+		return common.GeneralError(http.StatusBadRequest, response.InternalError, errMsg, nil, nil)
+	}
+
+	if installreq.Links == nil {
+		errMsg := "Invalid request, AuthorizedDevices links missing"
+		log.Error(errMsg)
+		return common.GeneralError(http.StatusBadRequest, response.InternalError, errMsg, nil, nil)
+	}
+	var serverURI string
+	var err error
+	var managerLink []string
+	linksMap := make(map[string]bool)
+	for _, serverIDs := range installreq.Links.Link {
+		serverURI = serverIDs.Oid
+		if strings.Contains(serverURI, "Systems") {
```

Comment:
Cant, we use a switch case to future proof the code? we won't have to add else conditions every time.